esptfa_arima/Authentication/services.py

```
@@ -148,45 +207,63 @@ def get_sections_dict() -> dict:
+                last_name = str(student.get("last_name")).strip()
+
+                if allowed_section:
+                    section = allowed_section
+                else:
+                    section_input = student.get("section")
+                    # Determine the section object
+                    section = sections_dict.get(section_input)
+                    if not section:
+                        raise DRFValidationError(
+                            f"Row {index + 1}: Section '{section_input}' does not exist"
+                        )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)